src/parser.ts

```diff
@@ -1870,6 +1871,39 @@ export class Parser extends DiagnosticEmitter {
     // implemented methods are virtual
     if (isInterface) flags |= CommonFlags.VIRTUAL;
 
+    var declareStart = 0;
+    var declareEnd = 0;
+    var contextIsAmbient = parent.is(CommonFlags.AMBIENT);
+    if (tn.peek() == Token.DECLARE) {
+      let state = tn.mark();
+      tn.next();
+      if (tn.peek() != Token.COLON) { // modifier
+        tn.discard(state);
+        if (isInterface) {
+          this.error(
+            DiagnosticCode._0_modifier_cannot_be_used_here,
+            tn.range(), "declare"
+          );
+        } else {
+          if (contextIsAmbient) {
+            this.error(
+              DiagnosticCode.A_declare_modifier_cannot_be_used_in_an_already_ambient_context,
+              tn.range()
+            ); // recoverable
+          } else {
+            flags |= CommonFlags.DECLARE | CommonFlags.AMBIENT;
+            declareStart = tn.tokenPos;
+            declareEnd = tn.pos;
+          }
+        }
+        if (!startPos) startPos = tn.tokenPos;
+      } else { // identifier
+        tn.reset(state);
+      }
+    } else if (contextIsAmbient) {
+      flags |= CommonFlags.AMBIENT;
+    }
+
```

Comment:
I'd prefer to make the `declare` changes a separate PR, since `declare`d class members would yield imports and it is not clear to me how these should work, in particular for fields.

Reply:
should I change the respective error message? (the one which mentions the declare keyword)

Reply:
There is not really a mechanism currently to make a `declare`d class member work in the first place, I think. What I imagine here is that this can still become a separate PR, so the parser understands the syntax, but would produce an error that ambient class members are not supported yet?